stan/math/opencl/rev/multiply.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef STAN_MATH_OPENCL_REV_MULTIPLY_HPP
+#define STAN_MATH_OPENCL_REV_MULTIPLY_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_generator/is_kernel_expression.hpp>
+#include <stan/math/opencl/multiply.hpp>
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/fun/value_of.hpp>
+#include <stan/math/rev/functor/reverse_pass_callback.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Matrix multiplication of two reverse mode matrices and/or kernel generator
+ * expressions.
+ * @tparam T_a type of first expression
+ * @tparam T_b type of second expression
+ * @param a first expression
+ * @param b second expression
+ * @return Matrix product of given arguments
+ */
+template <
+    typename T_a, typename T_b,
+    require_all_nonscalar_prim_or_rev_kernel_expression_t<T_a, T_b>* = nullptr,
+    require_any_var_t<T_a, T_b>* = nullptr>
+inline auto operator*(const T_a& a, const T_b& b) {
```

Comment:
I think the compiler needs access to the `multiply()` function not `*`

Reply:
Oh yeah, we need both of these.